x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/servicenow/auth_types/oauth.tsx

```diff
@@ -0,0 +1,228 @@
+      >
+        <EuiFieldText
+          fullWidth
+          isInvalid={isUserIdentifierInvalid}
+          readOnly={readOnly}
+          name="connector-servicenow-user-identifier"
+          value={userIdentifierValue || ''}
+          data-test-subj="connector-servicenow-user-identifier-form-input"
+          onChange={onChangeUserIdentifierInvalidEvent}
+          onBlur={() => {
+            if (!userIdentifierValue) {
+              onChangeUserIdentifierInvalidEvent();
+            }
+          }}
+          disabled={isLoading}
+        />
+      </EuiFormRow>
+      <EuiFormRow
+        id="connector-servicenow-keyid"
+        fullWidth
+        error={errors.keyId}
+        isInvalid={isKeyIdInvalid}
+        label={i18n.KEY_ID_LABEL}
+      >
+        <EuiFieldText
+          fullWidth
+          readOnly={readOnly}
+          isInvalid={isKeyIdInvalid}
+          name="connector-servicenow-jwt-key-id"
+          value={jwtKeyId || ''}
+          data-test-subj="connector-servicenow-jwt-key-id-form-input"
+          onChange={onChangeJWTKeyIdEvent}
+          onBlur={() => {
+            if (!jwtKeyId) {
+              onChangeJWTKeyIdEvent();
+            }
+          }}
+          disabled={isLoading}
+        />
+      </EuiFormRow>
+      <EuiFormRow
+        id="connector-servicenow-client-secret"
+        fullWidth
+        error={errors.clientSecret}
+        isInvalid={isClientSecretInvalid}
+        label={i18n.CLIENTSECRET_LABEL}
+      >
+        <EuiFieldPassword
+          fullWidth
+          isInvalid={isClientSecretInvalid}
+          readOnly={readOnly}
+          name="connector-servicenow-client-secret"
+          value={clientSecret || ''}
+          data-test-subj="connector-servicenow-client-secret-form-input"
+          onChange={onChangeClientSecretEvent}
+          onBlur={() => {
+            if (!clientSecret) {
+              onChangeClientSecretEvent();
+            }
+          }}
+          disabled={isLoading}
+        />
+      </EuiFormRow>
+      <EuiFormRow
+        id="connector-servicenow-private-key"
+        fullWidth
+        error={errors.privateKey}
+        isInvalid={isPrivateKeyInvalid}
+        label={i18n.PRIVATE_KEY_LABEL}
+      >
+        <EuiTextArea
+          fullWidth
+          readOnly={readOnly}
+          isInvalid={isPrivateKeyInvalid}
+          name="connector-servicenow-private-key"
+          value={privateKey || ''}
+          data-test-subj="connector-servicenow-private-key-form-input"
+          onChange={onChangePrivateKeyEvent}
+          onBlur={() => {
+            if (!privateKey) {
+              onChangePrivateKeyEvent();
+            }
+          }}
+          disabled={isLoading}
+        />
+      </EuiFormRow>
+      <EuiFormRow
+        id="connector-servicenow-private-key-password"
+        fullWidth
+        error={errors.privateKeyPassword}
+        label={i18n.PRIVATE_KEY_PASSWORD_LABEL}
+      >
+        <EuiFieldPassword
```

Comment:
Can we add helper text to this field that this is only required if the user has set a password on their private key?

Reply:
I had been thinking instead of a tooltip, adding helper text like there is underneath the ServiceNow URL field. `Include the full URL` is the helper text.

<img width="619" alt="Screen Shot 2022-05-16 at 8 40 26 AM" src="https://user-images.githubusercontent.com/13104637/168594456-9088a027-8106-4e89-899a-c5cc1ea0790d.png">
 
WDYT @mdefazio?

Reply:
@ymao1 I think that seems reasonable.  Did you have thoughts on the exact copy?
`Required if private key has a password`?

It might be worth trying to place this in the `labelAppend` prop ([doc link](https://elastic.github.io/eui/#/forms/form-layouts)).  This shows on the right side from the label.  Typically this is where we show 'Optional' or documentation links and feels more inline with that sort of content.  But I understand it might be a bit long to place there.

Reply:
I think the wording @ersin-erdal has is clear enough: `This is only required if you have set a password on your private key`